src/Components/SmartComponents/SystemDetailsPage/SystemDetails.js

```diff
@@ -3,17 +3,34 @@ import propTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { withRouter } from 'react-router-dom';
 import { ConnectedSystemCves } from '../SystemCves/SystemCves';
-import { EmptyStateExcludedSystem } from '../../PresentationalComponents/EmptyStates/EmptyStates';
+import {
+    EmptyStateExcludedSystem,
+    NotAuthorized
+} from '../../PresentationalComponents/EmptyStates/EmptyStates';
+import { Spinner } from '@redhat-cloud-services/frontend-components/Spinner';
+import { intl } from '../../../Utilities/IntlProvider';
+import messages from '../../../Messages';
 
-const SystemDetails = ({ entity, isOptOut, optOutSystemHandler, loaded }) => {
-    return entity && loaded && isOptOut ? (
-        <EmptyStateExcludedSystem buttonAction={() => optOutSystemHandler(false)} />
-    ) : (
+const SystemDetails = ({ entity, isOptOut, optOutSystemHandler, loaded, rbacPermissions, isRbacLoading }) => {
+    const {
+        canExport,
+        canReadVulnerabilityResults,
+        canEditPairStatus,
+        canReadExcludedIncluded
+    } = rbacPermissions;
+
+    let allowedCveActions = ['REMEDIATE', ...canEditPairStatus ? ['EDIT_STATUS'] : []];
+
+    return entity && loaded && isOptOut ? isRbacLoading ? <Spinner centered/>
+        : canReadVulnerabilityResults ? canReadExcludedIncluded ? (
+            <EmptyStateExcludedSystem buttonAction={() => optOutSystemHandler(false)}/>
+        ) : <NotAuthorized/> : <NotAuthorized title={intl.formatMessage(messages.notAuthorizedNoAccessForSystem)}/> : (
```

Comment:
Nesting 4 ternary statements is way too cluttered to be readable, can you please rework this condition? Don't be afraid to use if/else and multiple return statements if it helps.